src/test/java/ReceiptAnalysisTest.java

```diff
@@ -0,0 +1,111 @@
+
+package com.google.sps;
+
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.google.sps.data.AnalysisResults;
+import com.google.sps.servlets.ReceiptAnalysis;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({AnnotateImageResponse.class, AnnotateImageRequest.class,
+    BatchAnnotateImagesResponse.class, ByteString.class, EntityAnnotation.class, Feature.class,
+    Image.class, ImageAnnotatorClient.class, ReceiptAnalysis.class, URL.class})
+public final class ReceiptAnalysisTest {
+  private static final ByteString IMAGE_BYTES = ByteString.copyFromUtf8("byte string");
+  private static final String RAW_TEXT = "raw text";
+
+  @Mock private AnnotateImageRequest request;
+  @Mock private Feature feature;
+  @Mock private Image image;
+
+  @Mock(answer = Answers.RETURNS_SELF) private AnnotateImageRequest.Builder requestBuilder;
+  @Mock(answer = Answers.RETURNS_SELF) private Feature.Builder featureBuilder;
+  @Mock(answer = Answers.RETURNS_SELF) private Image.Builder imageBuilder;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void serveImageTextUrlReturnsAnalysisResults() throws IOException {
+    URL url = mock(URL.class);
+    InputStream inputStream = new ByteArrayInputStream(IMAGE_BYTES.toByteArray());
+    when(url.openStream()).thenReturn(inputStream);
+
+    mockStatic(ByteString.class);
+    when(ByteString.readFrom(inputStream)).thenReturn(IMAGE_BYTES);
+
+    when(imageBuilder.build()).thenReturn(image);
+    mockStatic(Image.class);
+    when(Image.newBuilder()).thenReturn(imageBuilder);
+
+    when(featureBuilder.build()).thenReturn(feature);
+    mockStatic(Feature.class);
+    when(Feature.newBuilder()).thenReturn(featureBuilder);
+
+    when(requestBuilder.build()).thenReturn(request);
+    mockStatic(AnnotateImageRequest.class);
+    when(AnnotateImageRequest.newBuilder()).thenReturn(requestBuilder);
```

Comment:
Reordering these blocks so that the stubbing goes in the same order as the execution of the methods would make it easier to follow.

For example, since `Image.newBuilder()` is called before `build()` in the ReceiptAnalysis class, the corresponding order is:
```
mockStatic(Image.class);
when(Image.newBuilder()).thenReturn(imageBuilder);
when(imageBuilder.build()).thenReturn(image);
```